api/src/main/java/com/clubobsidian/dynamicgui/api/effect/SoundWrapper.java

```diff
@@ -44,17 +44,32 @@ public SoundWrapper(@NotNull SoundData data) {
         this.data = Objects.requireNonNull(data);
     }
 
+    /**
+     * Gets the SoundData for the sound effect
+     *
+     * @return the SoundData object
+     */
     public SoundData getData() {
         return this.data;
     }
 
+    /**
+     * Plays the sound to the player
+     *
+     * @param player the player to play the sound to
+     */
     public void playSoundToPlayer(@NotNull PlayerWrapper<?> player) {
         Objects.requireNonNull(player);
         player.playSound(this.data);
     }
 
     public static class SoundData {
 
+        /**
+         * Creates a SoundData object from a string
+         * @param str The string to create the SoundData object from
+         * @return
```

Comment:
Add return comment, something along the lines of "Returns SoundData from the given string" should be fine